server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexAliasesService.java

```diff
@@ -244,4 +241,73 @@ private static void validateAliasTargetIsNotDSBackingIndex(ClusterState currentS
             );
         }
     }
+
+    /**
+     * A cluster state update task that applies the alias actions to the given cluster state.
+     */
+    record ApplyAliasActions(
+        MetadataIndexAliasesService aliasesService,
+        IndicesAliasesClusterStateUpdateRequest request,
+        ActionListener<AcknowledgedResponse> listener
+    ) implements ClusterStateTaskListener {
+
+        @Override
+        public void onFailure(Exception e) {
+            listener.onFailure(e);
+        }
+
+        ClusterStateAckListener getAckListener() {
+            return new ClusterStateAckListener() {
+                @Override
+                public boolean mustAck(DiscoveryNode discoveryNode) {
+                    return true;
+                }
+
+                @Override
+                public void onAllNodesAcked() {
+                    listener.onResponse(AcknowledgedResponse.TRUE);
+                }
+
+                @Override
+                public void onAckFailure(Exception e) {
+                    listener.onResponse(AcknowledgedResponse.FALSE);
+                }
+
+                @Override
+                public void onAckTimeout() {
+                    listener.onResponse(AcknowledgedResponse.FALSE);
+                }
+
+                @Override
+                public TimeValue ackTimeout() {
+                    return request.ackTimeout();
+                }
+            };
+        }
+
+        /**
+         * Handles the cluster state transition to a version that reflects the provided {@link AliasAction}s by calling
+         * the {@link MetadataIndexAliasesService#applyAliasActions(ClusterState, Iterable)}.
+         */
+        public ClusterState execute(ClusterState currentState) {
+            return aliasesService.applyAliasActions(currentState, request.actions());
+        }
+
+        static class Executor implements ClusterStateTaskExecutor<ApplyAliasActions> {
```

Comment:
Nit: I think I'd prefer this not to be an inner class of the task, maybe move it up a level?

Also could we just use `SimpleBatchedAckListenerTaskExecutor` here?

Reply:
Specifically, I think `Executor` should be a non-static inner class of `MetadataIndexAliasesService` so that there's no need to explicitly pass around any references to the `MetadataIndexAliasesService`. Something like this perhaps?

```
    class Executor extends SimpleBatchedAckListenerTaskExecutor<ApplyAliasActions> {
        @Override
        public Tuple<ClusterState, ClusterStateAckListener> executeTask(ApplyAliasActions applyAliasActions, ClusterState clusterState) throws Exception {
            return Tuple.tuple(applyAliasActions(clusterState, applyAliasActions.request().actions()), applyAliasActions.getAckListener());
        }
    }
```

Reply:
Ah yes, I wasn't aware of it. Nice

Reply:
I thought to just instantiate in the constructor and expose it to the test via a getter. I think as a class is minimal enough to have it there. What do you think?